supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,499 @@
+    val directoryPath = womFile.value.ensureSlashed
+    val directoryListFile = womFile.value.ensureUnslashed + ".list"
+    val dirDestinationPath = callRootPath.resolve(directoryPath).pathAsString
+    val listDestinationPath = callRootPath.resolve(directoryListFile).pathAsString
+
+    val (_, directoryDisk) = relativePathAndVolume(womFile.value, runtimeAttributes.disks)
+
+    // We need both the collection directory and the collection list:
+    List(
+      // The collection directory:
+      AwsBatchFileOutput(
+        makeSafeAwsBatchReferenceName(directoryListFile),
+        listDestinationPath,
+        DefaultPathBuilder.get(directoryListFile),
+        directoryDisk
+      ),
+      // The collection list file:
+      AwsBatchFileOutput(
+        makeSafeAwsBatchReferenceName(directoryPath),
+        dirDestinationPath,
+        DefaultPathBuilder.get(directoryPath + "*"),
+        directoryDisk
+      )
+    )
+  }
+
+  private def generateAwsBatchSingleFileOutputs(womFile: WomSingleFile): List[AwsBatchFileOutput] = {
+    val destination = callRootPath.resolve(womFile.value.stripPrefix("/")).pathAsString
+    val (relpath, disk) = relativePathAndVolume(womFile.value, runtimeAttributes.disks)
+    val output = AwsBatchFileOutput(makeSafeAwsBatchReferenceName(womFile.value), destination, relpath, disk)
+    List(output)
+  }
+
+  private def generateAwsBatchGlobFileOutputs(womFile: WomGlobFile): List[AwsBatchFileOutput] = {
+    val globName = GlobFunctions.globName(womFile.value)
+    val globDirectory = globName + "/"
+    val globListFile = globName + ".list"
+    val globDirectoryDestinationPath = callRootPath.resolve(globDirectory).pathAsString
+    val globListFileDestinationPath = callRootPath.resolve(globListFile).pathAsString
+
+    val (_, globDirectoryDisk) = relativePathAndVolume(womFile.value, runtimeAttributes.disks)
+
+    // We need both the glob directory and the glob list:
+    List(
+      // The glob directory:
+      AwsBatchFileOutput(makeSafeAwsBatchReferenceName(globDirectory), globDirectoryDestinationPath, DefaultPathBuilder.get(globDirectory + "*"), globDirectoryDisk),
+      // The glob list file:
+      AwsBatchFileOutput(makeSafeAwsBatchReferenceName(globListFile), globListFileDestinationPath, DefaultPathBuilder.get(globListFile), globDirectoryDisk)
+    )
+  }
+
+  lazy val monitoringParamName: String = AwsBatchJobPaths.AwsBatchMonitoringKey
+  lazy val localMonitoringLogPath: Path = DefaultPathBuilder.get(callPaths.monitoringLogFilename)
+  lazy val localMonitoringScriptPath: Path = DefaultPathBuilder.get(callPaths.monitoringScriptFilename)
+
+  // TODO: Determine the necessity for this monitoring stuff
+  lazy val monitoringScript: Option[AwsBatchInput] = {
+    // callPaths.workflowPaths.monitoringScriptPath map { path =>
+    //   AwsBatchFileInput(s"$monitoringParamName-in", path.pathAsString, localMonitoringScriptPath, workingDisk)
+    // }
```

Comment:
?

Reply:
This I'd like to leave in place until I finish the S3 integration. I'm unclear on how some of the monitoring stuff works in Cromwell, but as I move through the Centaur runs I'll see if this is necessary and either delete it or flesh out the method if that's ok.